flink-python/pyflink/fn_execution/coders.py

```diff
@@ -467,8 +478,8 @@ def from_proto(field_type):
     if field_type_name == type_name.LOCAL_ZONED_TIMESTAMP:
         timezone = pytz.timezone(os.environ['table.exec.timezone'])
         return LocalZonedTimestampCoder(field_type.local_zoned_timestamp_info.precision, timezone)
-    elif field_type_name == type_name.ARRAY:
-        return ArrayCoder(from_proto(field_type.collection_element_type))
+    elif field_type_name == type_name.BASIC_ARRAY:
```

Comment:
Should we also support PRIMITIVE_ARRAY here?

Reply:
In SQL/ Table API, the array type is the BasicArrayType,  PrimitiveArrayType is not considered.